addons/sale_management/views/sale_order_views.xml

```diff
@@ -81,6 +81,7 @@
                 <field name="sale_order_template_id" context="{'company_id': company_id}"
                     options="{'no_create': True, 'no_open': True}"
                     groups="sale_management.group_sale_order_template"
+                    domain = "['|', ('company_id', '=', False), ('company_id', '=', parent.company_id)]"
```

Comment:
The template is on the Sale Order, not on the sale order lines.  Therefore, the domain can be put directly in the python definition of the field, and use company_id instead of parent.company_id.